### ISSUE TITLE
[Linux] Double click fails to open project if path contains non-ASCII characters

### ISSUE THREAD

**Issue by unfa** [LMMS/lmms#1995]
I'm nurring KX Studio, the file manager is Dolphin (KDE).

When I doubleclick on a project file, LMMS opens but shows this dialog:

![lmms-doubleclick-project-load-fail](https://cloud.githubusercontent.com/assets/491704/7251191/4d0e85e0-e829-11e4-810b-a125e69133df.png)

And creates new session.

Opening the same file via File>Open dialog works fine.

**Comment by tresf** [comment]
At a glance looks related to #1325, where the latin1 version of the text is being processed rather than the utf8, etc

**Comment by softrabbit** [comment]
To rule out an oddly named folder, I'd like to see confirmation that:
* Dolphin shows the folder name OK
* Doubleclicking other files in the same folder works fine
* The folder name looks OK on the command line

**Comment by unfa** [comment]
Dolphin show the file name fine.
Doubleclicking other files (WAV for example in the same dir) works.
Running form comandline:

    lmms /unfa/Projects/Mieczysław Szymasiewicz/Krainy Oniryczne/Muzyka/Zona/unfa-Zona-01.mmpz

Failed

    cd /unfa/Projects/Mieczysław Szymasiewicz/Krainy Oniryczne/Muzyka/Zona/
    lmms ./unfa-Zona-01.mmpz

Worked

**Comment by softrabbit** [comment]
>I guess the Polish diacritics like "ł" are the problem on some level.
Yes, looks like ```QString```s constructed from ```char *``` are converted using ```fromAscii()```. Should probably use ```fromLocal8Bit()``` instead.

**Comment by tresf** [comment]
Thanks for the sample text.  It is being triggered here:  [`DataFile.cpp#L126`](https://github.com/LMMS/lmms/blob/master/src/core/DataFile.cpp#L126) by improper encoding of the `dark L` or `L with stroke`, i.e. `ł`.

Had this been a UTF8 issue, the solutions proposed seemed simple, but to my surprise, I've had no luck getting them to fix the issue.

So I dug a bit deeper and stumbled upon this.... our main constructor casts the char* to QString which may be causing it per:

> The default `char* => QString` conversion uses Latin-1, which is not what you want for file names.
>
> Source:  http://stackoverflow.com/questions/1641994/qdir-and-qdiriterator-ignore-files-with-non-ascii-filenames

The proposal is to instead use:

```cpp
QApplication app(argc, argv);

QStringList args = app.arguments();

for(...)
```

But this changes our `main.cpp` logic a bit, so it would require a bit of code refactoring.

**Comment by tresf** [comment]
Looks like @softrabbit and I did similar work here.  I'd recommend we don't override the default encoding settings, but rather use the Qt recommended approach for reading application parameters, as if we force UTF8, we may run into issues with Windows-CP1252, etc.

-Tres

**Comment by softrabbit** [comment]
The ```fromLocal8Bit()``` idea came straight from here: http://doc.qt.io/qt-5/qcoreapplication.html#arguments so it should be pretty much safe.

**Comment by tresf** [comment]
Good catch, that seems to work.

![image](https://cloud.githubusercontent.com/assets/6345473/7259714/49131302-e833-11e4-9ff4-b1f68152bffa.png)

We'll need to patch it into several places due to the command line rendering, etc.

**Comment by tresf** [comment]
@softrabbit can you take a look at #1998 and see if I missed anything?

**Comment by tresf** [comment]
Closed via #1998. Added to release notes under "Fixes" section.
